### PR TITLE
last_date_to_pay field can be none

### PR DESCRIPTION
**Pull request by SL1994-design**
### **User description**
add none as default values to some fields

## How did you test these changes? 

**Closing issues**

___

### **PR Type**
enhancement

___

### **Description**
- Updated the `ElectricBill` dataclass to allow the `last_date_to_pay` field to be `None` by using `Optional` typing and setting a default value of `None`.
- Modified the `ResponseDescriptor` and `ResponseWithDescriptor` dataclasses to set default values of `None` for `code`, `description`, and `data` fields, enhancing the flexibility of these models.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>electric_bill.py</strong><dd><code>Allow `last_date_to_pay` field to be None</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

iec_api/models/electric_bill.py

<li>Added <code>Optional</code> typing to <code>last_date_to_pay</code>.<br> <li> Set default value of <code>last_date_to_pay</code> to <code>None</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/GuyKh/py-iec-api/pull/159/files#diff-412f01907575556be9e7586efdf0cd9c30236ea057f0797f5469b6c8892adac0">+3/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>response_descriptor.py</strong><dd><code>Set default None values for response fields</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

iec_api/models/response_descriptor.py

<li>Set default values of <code>code</code> and <code>description</code> to <code>None</code>.<br> <li> Set default value of <code>data</code> to <code>None</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/GuyKh/py-iec-api/pull/159/files#diff-c5ce5614225adba33d5889603c10e82e15a3d229faaacfdcf2008ec7cade0036">+3/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

### CODE REVIEW

File: iec_api/models/response_descriptor.py [hunk 1]
```diff
@@ -11,8 +11,8 @@ class ResponseDescriptor(DataClassDictMixin):
     """Response Descriptor"""
 
     is_success: bool = field(metadata=field_options(alias="isSuccess"))
-    code: Optional[str]
-    description: Optional[str]
+    code: Optional[str] = None
```
Comment (GuyKh) [inline iec_api/models/response_descriptor.py:14]: No reason for setting default values, they'll be `None` by default

**Comment by GuyKh** [comment]
@SL1994-design - please fix the lint comments

File: iec_api/models/response_descriptor.py (hunk 1 above)
Comment (SL1994-design) [reply iec_api/models/response_descriptor.py:14]: I had some errors without this added line. for example: `errorMessage": "Field \"response_descriptor\" of type ResponseDescriptor in ResponseWithDescriptor has invalid value {'isSuccess': True, 'code': '00'}",`

Comment (GuyKh) [reply iec_api/models/response_descriptor.py:14]: When trying to parse the response:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<string>", line 6, in __mashumaro_from_dict__
mashumaro.exceptions.MissingField: Field "data" of type Optional[Any] is missing in ResponseWithDescriptor instance
```

This means that `data` field (and not `code`) can be None.
Which you already added.

**Comment by GuyKh** [comment]
> @SL1994-design - please fix the lint comments

```
iec_api/models/electric_bill.py:55:61: W292 [*] No newline at end of file
   |
55 | decoder = BasicDecoder(ResponseWithDescriptor[ElectricBill])
   |                                                              W292
   |
   = help: Add trailing newline

iec_api/models/response_descriptor.py:38:29: W292 [*] No newline at end of file
   |
37 |     response_descriptor: ResponseDescriptor = field(metadata=field_options(alias=RESPONSE_DESCRIPTOR_FIELD))
38 |     data: Optional[T] = None
   |                              W292
   |
   = help: Add trailing newline

```

Please add newlines at the end of the files

**Comment by GuyKh** [comment]
@SL1994-design - please do it for both files.

**Review by GuyKh — CHANGES_REQUESTED** [review]
Please see lint errors.

**Pull request merged**